src/main/java/cs2103/v15_1j/jimjim/ui/TaskEventRowFactory.java

```diff
@@ -0,0 +1,492 @@
+			addNoEventTaskLabel(selectedDate);
+		}
+
+	}
+
+	public int showTaskEventsOnDate(LocalDate date){
+		int noOfEvents = addEventOnDate(date);
+		int noOfTasks = addDeadlineTaskOnDate(date);
+
+		rowNo += (noOfEvents+noOfTasks);
+
+		Label emptyLabel = new Label("");
+		pane.add(emptyLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+
+		if ((noOfEvents+noOfTasks) == 0){
+			pane.getChildren().remove(emptyLabel);
+		}
+
+		return noOfEvents + noOfTasks;
+	}
+
+	public int showOverdue(){
+		int noOfOverdue = 0;
+
+		for(Event e: masterList.getEventsList()){
+			if(checkOverdue(e)){
+				add(e);
+				noOfOverdue++;
+			}
+		}
+
+		for(DeadlineTask t: masterList.getDeadlineTasksList()){
+			if(checkOverdue(t)){
+				add(t);
+				noOfOverdue++;
+			}
+		}
+
+		if(noOfOverdue == 0){
+			Label noOverdueLabel = new Label("No Overdue Tasks or Events");
+			noOverdueLabel.getStyleClass().add("no-overdue-label");
+			pane.add(noOverdueLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+		} else {
+			rowNo += noOfOverdue;
+		}
+
+		return noOfOverdue;
+	}
+
+	public boolean showFloatingTasks(boolean showCompleted){
+		boolean hasCompleted = false;
+
+		for(FloatingTask t: masterList.getFloatingTasksList()){
+			if(!t.getCompleted()){
+				rowNo++;
+				add(t);
+			}
+			else {
+				hasCompleted = true;
+			}
+		}
+
+		if(showCompleted){
+			for(FloatingTask t: masterList.getFloatingTasksList()){
+				if(t.getCompleted()){
+					rowNo++;
+					add(t);
+				}
+			}
+		}
+
+		return hasCompleted;
+	}
+
+	public void showSearchResults(){
+		pane.getChildren().clear();
+		rowNo = new Integer(0);
+
+		Label titleLabel = new Label("Search Results");
+		titleLabel.getStyleClass().add("header");
+		pane.add(titleLabel, 0, rowNo, NO_OF_COLUMNS, 1);
+
+		LocalDate firstDate = getFirstDate();
+
+		if(firstDate == null && masterList.getFloatingTasksList().isEmpty()){
+			Label emptyLabel = new Label("No results match your search term");
+			emptyLabel.getStyleClass().add("red-label");
+			pane.add(emptyLabel, 0, 1, NO_OF_COLUMNS, 1);
+		}
+		else if (firstDate != null) {
+			LocalDate tempDate = firstDate;
+
+			while(!tempDate.isAfter(getLastDate())){
+				addTaskEventsOnDate(tempDate);
+				tempDate = tempDate.plusDays(1);
+			}
+		}
+
+		for(FloatingTask task: masterList.getFloatingTasksList()){
+			add(task);
+		}
+	}
+
+	private int addTaskEventsOnDate(LocalDate date){
+		Label dateLabel = new Label(date.format(dateFmt));
+		dateLabel.getStyleClass().add("date-label");
+		pane.add(dateLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+
+		int noOfEvents = addEventOnDate(date);
+		int noOfTasks = addDeadlineTaskOnDate(date);
+
+		rowNo += (noOfEvents+noOfTasks);
+
+		Label emptyLabel = new Label("");
+		pane.add(emptyLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+
+		if ((noOfEvents+noOfTasks) == 0){
+			pane.getChildren().remove(dateLabel);
+			pane.getChildren().remove(emptyLabel);
+		}
+
+		return noOfEvents + noOfTasks;
+	}
+
+	private int addTaskEventsOnDate(LocalDate date, LocalDate selectedDate){
```

Comment:
what's the diff between this and the previous one beside adding the no task message at the end?